lib/winapi/sync.c

```diff
@@ -1,11 +1,145 @@
 #include <synchapi.h>
 #include <assert.h>
+#include <stdatomic.h>
 #include <stdbool.h>
 #include <processthreadsapi.h>
 #include <winbase.h>
 #include <winerror.h>
 #include <xboxkrnl/xboxkrnl.h>
 
+static BOOL WINAPI condvar_init (PINIT_ONCE InitOnce, PVOID Parameter, PVOID *Context)
+{
+    PCONDITION_VARIABLE ConditionVariable = Parameter;
+    NTSTATUS status;
+    status = NtCreateEvent(&ConditionVariable->eventHandles[0], NULL, SynchronizationEvent, FALSE);
+    assert(NT_SUCCESS(status));
+    status = NtCreateEvent(&ConditionVariable->eventHandles[1], NULL, NotificationEvent, FALSE);
+    assert(NT_SUCCESS(status));
+    return true;
+}
+
+VOID InitializeConditionVariable (PCONDITION_VARIABLE ConditionVariable)
+{
+    ConditionVariable->initOnce.Ptr = 0;
+    ConditionVariable->eventHandles[0] = INVALID_HANDLE_VALUE;
+    ConditionVariable->eventHandles[1] = INVALID_HANDLE_VALUE;
```

Comment:
Why is this necessary if `condvar_init` does it anyway? It should only have to be set in the case where `InitOnceExecuteOnce` fails.

Reply:
See https://github.com/XboxDev/nxdk/pull/274#discussion_r424076346 ; I'll mark this conversation as resolved.